### PR TITLE
[Darwin][Input] Extend GCController support to osx

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Support GCController API for OSX

## Motivation and Context
Bluetooth xbox game controller on current osx, dpad isnt detected working. GCController API has been available to OSX since 10.9 (in limited ways) and looks to be reasonably feature complete between osx/ios/tvos now.

Essentially this PR just renames some classes, and opens checks for osx. No functional changes to the code were required to get working.

Requires a change to peripheral.joystick (relocate darwin_embedded xml layouts to a generic GCController folder). Will PR this change once we get into v20 dev cycle

## How Has This Been Tested?
OSX 10.15.7 (x86_64), Big Sur 11.0.1 (Apple Silicon)

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [X] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by garbear** [comment]
The changes here look good. Should we drop cocoa support from peripheral.joystick?

**Comment by fuzzard** [comment]
That would be my intention long term. To do that though, it will enforce minimum OSX versioning. I've still to look into the full ramifications of that

**Comment by garbear** [comment]
Got it, core conflicts are valid reasons to do game controllers in the addon.

File: xbmc/platform/darwin/peripherals/Input_Gamecontroller.mm [hunk 1]
```diff
@@ -155,14 +155,14 @@ - (void)registerChangeHandler:(GCController*)controller
     CLog::Log(LOGDEBUG, "INPUT - GAMECONTROLLER: microGamepad changehandler added");
     [self microValueChangeHandler:controller];
   }
-  if (@available(iOS 13.0, tvOS 13.0, *))
+  if (@available(iOS 13.0, tvOS 13.0, macOS 10.15, *))
```
Comment (kambala-decapitator) [inline xbmc/platform/darwin/peripherals/Input_Gamecontroller.mm:158]: in this case correct way would be to use `NSProcessInfo` to determine OS version at runtime, as `@available` is used for new API checks

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]

File: xbmc/platform/darwin/peripherals/Input_Gamecontroller.mm (hunk 1 above)
Comment (fuzzard) [reply xbmc/platform/darwin/peripherals/Input_Gamecontroller.mm:158]: As discussed, seems cleaner to use @available for now.

**Review by garbear — APPROVED** [review]

**Comment by fuzzard** [comment]
@garbear I'll look to merge this once https://github.com/xbmc/peripheral.joystick/pull/192 goes in.

Are you able to tag a release of peripheral.joystick as well.

**Comment by garbear** [comment]
@fuzzard merged https://github.com/xbmc/peripheral.joystick/pull/192 and tagged a version. the release will work its way through CI

**Pull request merged**